svc/sentinel/engine/ratelimit/headers.go

```
@@ -0,0 +1,56 @@
+// writePolicyRateLimitHeaders sets X-RateLimit-* headers on the response.
+// Multiple policies (KeyAuth, RateLimit) write to the same standard headers,
+// so we only overwrite when the current result is more restrictive.
+// Retry-After is only added on denial.
+func writePolicyRateLimitHeaders(w http.ResponseWriter, resp rl.RatelimitResponse, clk clock.Clock) {
+	h := w.Header()
+
+	if !shouldOverwrite(h, resp) {
+		return
+	}
+
+	h.Set("X-RateLimit-Limit", strconv.FormatInt(resp.Limit, 10))
```

golangci-lint (2.11.4)

[error] non-canonical header "X-RateLimit-Limit", instead use: "X-Ratelimit-Limit"

(canonicalheader)